diff -urN Atari800/src/atari_x11.c Atari800-shm/src/atari_x11.c
--- Atari800/src/atari_x11.c	Sun Nov 11 16:39:09 2001
+++ Atari800-shm/src/atari_x11.c	Sun Dec  2 19:19:29 2001
@@ -2605,8 +2605,8 @@
 	first_y = ATARI_HEIGHT;
 	last_y = -1000;
 
-	if( invisible || !draw_display )   goto after_screen_update;  /* mmm */
-        if(image->bits_per_pixel == 32)
+	if (invisible)   goto after_screen_update;  /* mmm */
+        if (image->bits_per_pixel == 32)
         {
                 ULONG *ptr = (ULONG *) image->data;
                 UBYTE *ptr2 = screen;
